libstuff/SDeburr.cpp

```diff
@@ -0,0 +1,137 @@
+#include "SDeburr.h"
+
+#include <bit>
+#include <cstring>
+#include <string>
+
+#include <libstuff/libstuff.h>
+#include <libstuff/sqlite3.h>
+
+using namespace std;
+
+const char* SDeburr::unicodeToAscii(uint32_t codepoint) {
+    if (codepoint >= 0x00C0 && codepoint <= 0x017F) {
+        return SDeburr::UNICODE_TO_ASCII_MAP[codepoint - 0x00C0];
+    }
+
+    if (codepoint == 0x1E9E) {
+        return "SS"; // ẞ
+    }
+
+    return nullptr;
+}
+
+string SDeburr::deburr(const string& input) {
+    const unsigned char* input_bytes = reinterpret_cast<const unsigned char*>(input.c_str());
+    const size_t len = input.size();
+    string result;
+    result.reserve(len);
+    size_t i = 0;
+    while (i < len) {
+        // Count the leading ones in the current byte
+        // 0 leading ones -> ASCII character
+        // 1 leading one -> continuation byte
+        // 2-4 leading ones -> 2, 3, or 4-byte sequence
+        // 5+ leading ones -> invalid UTF-8
+        const int numLeadingOnes = countl_one(input_bytes[i]);
+
+        // Most common case: regular ASCII text
+        if (numLeadingOnes == 0) {
+            result.append(reinterpret_cast<const char*>(input_bytes + i), 1);
```

Comment:
I think this is clearer as just result.append(1, input_bytes[i]);

Reply:
updated